src/java.base/share/classes/java/util/random/RandomGenerator.java

```diff
@@ -0,0 +1,1634 @@
+    }
+
+    /**
+     * Returns a stream producing the given {@code streamSize} number of
+     * (pseudo)randomly chosen {@code long} values.
+     *
+     * @param streamSize the number of values to generate
+     *
+     * @return a stream of (pseudo)randomly chosen {@code long} values
+     *
+     * @throws IllegalArgumentException if {@code streamSize} is
+     *         less than zero
+     *
+     * @implNote The default implementation produces a sequential stream
+     * that repeatedly calls {@link RandomGenerator#nextLong() nextLong}().
+     */
+    default LongStream longs(long streamSize) {
+        RandomSupport.checkStreamSize(streamSize);
+
+        return longs().limit(streamSize);
+    }
+
+    /**
+     * Returns a stream producing the given {@code streamSize} number of
+     * (pseudo)randomly chosen {@code long} values, where each value is between
+     * the specified origin (inclusive) and the specified bound (exclusive).
+     *
+     * @param streamSize the number of values to generate
+     * @param randomNumberOrigin the least value that can be produced
+     * @param randomNumberBound the upper bound (exclusive) for each value produced
+     *
+     * @return a stream of (pseudo)randomly chosen {@code long} values, each between
+     *         the specified origin (inclusive) and the specified bound (exclusive)
+     *
+     * @throws IllegalArgumentException if {@code streamSize} is
+     *         less than zero, or {@code randomNumberOrigin}
+     *         is greater than or equal to {@code randomNumberBound}
+     *
+     * @implNote The default implementation produces a sequential stream that repeatedly
+     *            calls {@link RandomGenerator#nextLong(long, long) nextLong}(randomNumberOrigin, randomNumberBound).
+     */
+    default LongStream longs(long streamSize, long randomNumberOrigin,
+                long randomNumberBound) {
+        RandomSupport.checkStreamSize(streamSize);
+        RandomSupport.checkRange(randomNumberOrigin, randomNumberBound);
+
+        return longs(randomNumberOrigin, randomNumberBound).limit(streamSize);
+    }
+
+    /**
+     * Returns a (pseudo)randomly chosen {@code boolean} value.
+     *
+     * <p> The default implementation tests the high-order bit (sign bit) of a
+     * value produced by {@link RandomGenerator#nextInt() nextInt}(), on the
+     * grounds that some algorithms for pseudorandom number generation produce
+     * values whose high-order bits have better statistical quality than the
+     * low-order bits.
+     *
+     * @return a (pseudo)randomly chosen {@code boolean} value
+     */
+    default boolean nextBoolean() {
+        return nextInt() < 0;
+    }
+
+    /**
+     * Fills a user-supplied byte array with generated byte values
+     * (pseudo)randomly chosen uniformly from the range of values between -128
+     * (inclusive) and 255 (inclusive).
```

Comment:
Should this be
between -128 (inclusive) and **127** (inclusive)

Reply:
Thanks. Will update.